v2/tools/generator/internal/codegen/pipeline/delete_generated_code.go

```diff
@@ -64,26 +64,56 @@ func deleteGeneratedCodeByPattern(ctx context.Context, globPattern string) error
 		return errors.Wrapf(err, "error globbing files with pattern %q", globPattern)
 	}
 
-	var errs []error
-	for _, file := range files {
+	// We treat files as a queue of files needing deletion
+	// If we have an error deleting a file, it might be due to a transient lock on the file (e.g. from an editor or
+	// security software); if this happens, we requeue the file so we retry the deletion later
+	// To avoid getting stuck in an infinite loop, we keep track of the number of consequetive deletion failures; if
+	// this exceeds the size of the queue, we give up and return an error
+
+	consecutiveDeleteFailures := 0
+	errorsSeen := make(map[string]error, len(files))
```

Comment:
Just curious -- is this in response to you actually having seen this be an issue? I've never experienced it locally.

Reply:
Yup. Hits me all the time - Microsoft Defender locks the file and doesn't release the lock in time for the deletion to work.